UDPServer/ServerContextRepo.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Threading.Tasks;
+using UDPCommGUI;
+using System.Linq;
+using System;
+
+namespace Sice.PoC.UDPServer
+{
+    public class ServerContextRepo
+    {
+        public ServerContext context { get; set; }
+        public ServerContextRepo(ServerContext context)
+        {
+            this.context = context;
+        }
+
+        public async Task<int> AddReceivedMessage(ReceivedMessage message)
+        {
+            context.ReceivedMessages.Add(message);
+            return await context.SaveChangesAsync();
```

Comment:
Slight shortcut here I learnt from someone else.

if there is only one await and it's part of the return statement, you can get rid of the async and just return the Task

https://stackoverflow.com/questions/23382851/unnecessary-async-await-when-await-is-last

https://riptutorial.com/csharp/example/16320/returning-a-task-without-await

Essentially it adds an additional overhead - I can't find the exact article